contrib/jinzhu/gorm/gorm_test.go

```diff
@@ -81,3 +87,235 @@ func TestPostgres(t *testing.T) {
 	}
 	sqltest.RunAll(t, testConfig)
 }
+
+type Product struct {
+	gorm.Model
+	Code  string
+	Price uint
+}
+
+func TestWithContext(t *testing.T) {
+	assert := assert.New(t)
+	sqltrace.Register("postgres", &pq.Driver{})
+	db, err := Open("postgres", "postgres://postgres:postgres@127.0.0.1:5432/postgres?sslmode=disable")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+	db.AutoMigrate(&Product{})
+
+	s1, ctx1 := tracer.StartSpanFromContext(context.Background(), "http.request",
+		tracer.ServiceName("fake-http-server"),
+		tracer.SpanType(ext.SpanTypeWeb),
+	)
+	defer s1.Finish()
+	s2, ctx2 := tracer.StartSpanFromContext(context.Background(), "http.request",
+		tracer.ServiceName("fake-http-server"),
+		tracer.SpanType(ext.SpanTypeWeb),
+	)
+	defer s2.Finish()
+
+	db1 := WithContext(ctx1, db)
+	db2 := WithContext(ctx2, db)
+
+	_, ok := db.Get(gormContextKey)
```

Comment:
It seems a bit like these assertions are the result of taking the "shortcut" with this test. Can we instead trigger some tracing and assert that the appropriate spans and correct hierarchy is created? We probably don't need 2 contexts for that. I assume that's what you intend to test.

Reply:
Yeah, this test came from earlier in the implementation to make sure I understood how `Set` and `Get` worked on a `gorm.DB`.
The entire test could be removed if you like. It's basically just confirming that `gorm.DB.Set()` returns a unique object instead of modifying the original.

Reply:
If the use case is covered in the other tests implicitly, let's remove it.